chia/wallet/did_wallet/did_wallet.py

```diff
@@ -414,22 +418,26 @@ def puzzle_for_pk(self, pubkey: bytes) -> Program:
         innerpuz = did_wallet_puzzles.create_innerpuz(
             pubkey, self.did_info.backup_ids, self.did_info.num_of_backup_ids_needed
         )
-        did = self.did_info.my_did
-        return did_wallet_puzzles.create_fullpuz(innerpuz, did)
+        if self.did_info.origin_coin is not None:
```

Comment:
This is a good check.

Maybe later try to move this conversion closer to where `self.did_info.origin_coin` is assigned.